src/sources/datadog_agent/mod.rs

```diff
@@ -188,53 +189,55 @@ impl SourceConfig for DatadogAgentConfig {
-
-                    // Syslog deserializer allows for arbritrary "structured data" that can overwrite
-                    // existing fields, similar to the JSON deserializer.
-                    //
-                    // See also: https://datatracker.ietf.org/doc/html/rfc5424#section-6.3
-                    #[cfg(feature = "sources-syslog")]
-                    DeserializerConfig::Syslog => self.decoding.schema_definition(log_namespace),
-
-                    DeserializerConfig::Native => self.decoding.schema_definition(log_namespace),
-                    DeserializerConfig::NativeJson => {
-                        self.decoding.schema_definition(log_namespace)
-                    }
-                    DeserializerConfig::Gelf => self.decoding.schema_definition(log_namespace),
-                }
-            }
-            LogNamespace::Vector => self
-                .decoding
-                .schema_definition(log_namespace)
-                .with_metadata_field("message", Kind::bytes())
-                .with_metadata_field("status", Kind::bytes())
-                .with_metadata_field("timestamp", Kind::timestamp())
-                .with_metadata_field("hostname", Kind::bytes())
-                .with_metadata_field("service", Kind::bytes())
-                .with_metadata_field("ddsource", Kind::bytes())
-                .with_metadata_field("ddtags", Kind::bytes()),
-        };
+        let source_name = "datadog_agent";
+
+        let definition = self
+            .decoding
+            .schema_definition(global_log_namespace.merge(self.log_namespace))
+            .with_source_metadata(
+                source_name,
+                "message",
+                "message",
+                Kind::bytes(),
+                Some("message"),
+            )
+            .with_source_metadata(
+                source_name,
+                "status",
+                "status",
+                Kind::bytes(),
+                Some("severity"),
+            )
+            .with_source_metadata(
+                source_name,
+                "timestamp",
+                "timestamp",
+                Kind::timestamp(),
+                Some("timestamp"),
+            )
+            .with_source_metadata(
+                source_name,
+                "hostname",
+                "hostname",
+                Kind::bytes(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                source_name,
+                "service",
+                "service",
+                Kind::bytes(),
+                Some("service"),
+            )
+            .with_source_metadata(
+                source_name,
+                "ddsource",
+                "ddsource",
+                Kind::bytes(),
+                Some("source"),
+            )
+            .with_source_metadata(source_name, "ddtags", "ddtags", Kind::bytes(), Some("tags"))
```

Comment:
:table_flip: just this one that doesn't get line breaks?

Reply:
It would look a lot nicer with longer lines :disappointed:

Reply:
I'd be happy if it was just all the same 😆

Reply:
I tried adding line breaks and they get removed.